src/main/java/com/woowacamp/soolsool/core/liquor/repository/LiquorQueryDslRepository.java

```diff
@@ -0,0 +1,96 @@
+package com.woowacamp.soolsool.core.liquor.repository;
+
+import static com.woowacamp.soolsool.core.liquor.domain.QLiquor.liquor;
+
+import com.querydsl.core.types.Projections;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import com.woowacamp.soolsool.core.liquor.domain.LiquorBrew;
+import com.woowacamp.soolsool.core.liquor.domain.LiquorRegion;
+import com.woowacamp.soolsool.core.liquor.domain.LiquorStatus;
+import com.woowacamp.soolsool.core.liquor.domain.vo.LiquorBrand;
+import com.woowacamp.soolsool.core.liquor.dto.LiquorElementResponse;
+import com.woowacamp.soolsool.core.liquor.dto.LiquorSearchCondition;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class LiquorQueryDslRepository {
+
+    private final JPAQueryFactory queryFactory;
+
+    public List<LiquorElementResponse> getList(
+        final LiquorSearchCondition condition,
+        final Pageable pageable,
+        final Long cursorId
+    ) {
+        return queryFactory.select(
+                Projections.constructor(
+                    LiquorElementResponse.class,
+                    liquor
+                )
+            )
+            .from(liquor)
+            .where(
+                eqRegion(condition.getLiquorRegion()),
+                eqBrew(condition.getLiquorBrew()),
+                eqStatus(condition.getLiquorStatus()),
+                eqBrand(condition.getBrand()),
+                cursorId(cursorId)
+            )
+            .orderBy(liquor.id.desc())
+            .limit(pageable.getPageSize())
+            .fetch();
+    }
+
+    @Cacheable(value = "liquorsFirstPage")
+    public List<LiquorElementResponse> getCachedList(
+        final Pageable pageable
+    ) {
+        return getList(
+            LiquorSearchCondition.nullObject(),
+            pageable,
+            null
+        );
+    }
+
+    private BooleanExpression eqRegion(final Optional<LiquorRegion> liquorRegion) {
+        if (liquorRegion.isEmpty()) {
+            return null;
+        }
+        return liquor.region.eq(liquorRegion.get());
+    }
```

Comment:
```suggestion
    private BooleanExpression equalsRegion(final Optional<LiquorRegion> liquorRegion) {
        return liquorRegion.map(liquor.region::eq).orElse(null);
    }
```

equals라는 명확한 이름과 적은 코드량으로 변경해봤습니다.